lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -720,33 +709,35 @@ def _cleanup_bootstrap_conf_if_applies(self) -> None:
         """Remove some conf props in the CM nodes that contributed to the cluster bootstrapping."""
         self.opensearch_config.cleanup_bootstrap_conf()
 
-    def _restart_unit_with_conf_if_needed(self):
-        """Trigger a restart event on the current unit if a recomputed conf was passed."""
-        node_conf_update_app = self.peers_data.get_object(Scope.APP, "update-node-config")
-        if not node_conf_update_app:
+    def _reconfigure_and_restart_unit_if_needed(self):
+        """Reconfigure the current unit if a new config was computed for it, then restart."""
+        nodes_config = self.peers_data.get_object(Scope.APP, "nodes_config")
+        if not nodes_config:
             return
 
-        node = Node.from_dict(node_conf_update_app)
-        if node.name != self.unit_name:
+        new_node_conf = nodes_config.get(self.unit_name)
+        if not new_node_conf:
             return
```

Comment:
if we don't have a config for the member, maybe it is become removed from the configuration.
should we stop OpenSearch?

Reply:
Good point, I think we should restart it in that case, and its conf will be auto-generated

Reply:
what if `config exists + no node in config + opensearch is up`